estate/models/est_estate_property.py

```diff
@@ -0,0 +1,95 @@
+    date_availability = fields.Date('Available From', copy=False,
+                                    default=fields.Date.add(fields.Date.today(), months=3))
+    expected_price = fields.Float(required=True)
+    selling_price = fields.Float(readonly=True, copy=False, compute='_compute_selling_price')
+    bedrooms = fields.Integer(default=2)
+    living_area = fields.Integer('Living Area (sqm)')
+    facades = fields.Integer()
+    garage = fields.Boolean()
+    garden = fields.Boolean()
+    garden_area = fields.Integer('Garden Area (sqm)')
+    garden_orientation = fields.Selection(
+        selection=[('north', 'North'), ('south', 'South'), ('east', 'East'), ('west', 'West')])
+    active = fields.Boolean(default=True)
+    state = fields.Selection(
+        selection=[('new', 'New'), ('offer received', 'Offer Received'), ('offer accepted', 'Offer Accepted'),
+                   ('sold', 'Sold'), ('cancelled', 'Cancelled')],
+        default='new')
+    type_id = fields.Many2one("est.estates.property.type", string="Property Type")
+    salesman = fields.Many2one("res.users", default=lambda self: self.env.user)
+    buyer = fields.Many2one("res.partner", copy=False, readonly=True)
+    tag_ids = fields.Many2many("est.estates.property.tag", string="Tags")
+    offer_ids = fields.One2many("est.estates.property.offer", "property_id", string="Offers")
+    total_area = fields.Integer('Total Area (sqm)', compute='_set_area', readonly=True)
+    best_price = fields.Float(compute='_compute_best_price')
+
+    _sql_constraints = [
+        ('Check_expected_price', 'CHECK(expected_price > 0)',
+         'The expected price should be greater than 0!')
+    ]
+    _sql_constraints = [
+        ('Check_selling_price', 'CHECK(selling_price >= 0)',
+         'The selling price should be greater or equal to 0')
+    ]
+    # private functions
+
+    @api.depends('living_area', 'garden_area')
+    def _set_area(self):
+        for record in self:
+            record.total_area = record.living_area + record.garden_area
+
+    @api.depends('offer_ids.price')
+    def _compute_best_price(self):
+        self.best_price = 0
+        for offer in self.offer_ids:
+            if self.best_price < offer.price:
+                self.best_price = offer.price
+
+    @api.depends('offer_ids.status')
+    def _compute_selling_price(self):
+        self.selling_price = 0
+        self.buyer = ''
+        for offer in self.offer_ids:
+            if offer.status == 'accepted':
+                self.selling_price = offer.price
+                self.buyer = offer.partner_id
+
+    @api.onchange('garden')
+    def _onchange_garden(self):
+        if self.garden:
+            self.garden_orientation = 'north'
+            self.garden_area = 10
+        else:
+            self.garden_orientation = ''
+            self.garden_area = 0
+
+    # public functions
+
+    def action_property_sold(self):
+        for record in self:
+            if record.state == 'cancelled':
```

Comment:
same for record, maybe something more obvious (even if here it doesn't matter that much ofc as the purpose of the method is really clear)